dftd3/dftd3.py

```diff
@@ -106,28 +105,28 @@ def d3(
     for j in range(MAX_ELEMENTS):
         mxc.append(0)
         for k in range(natom):
-            if atoms[k] > -1:
+            if charges[k] > -1:
                 for l in range(MAX_CONNECTIVITY):
                     if isinstance(C6AB[j][j][l][l], (list, tuple)):
                         if C6AB[j][j][l][l][0] > 0:
                             mxc[j] = mxc[j] + 1
                 break
 
     # Coordination number based on covalent radii
-    cn = ncoord(natom, atoms, coordinates)
+    cn = ncoord(natom, charges, coordinates)
 
     # compute C6, C8, and C10 coefficietns from tabulated values (in C6AB) and fractional coordination
     for j in range(natom):
         # C6 coefficient
-        C6jj = getc6(C6AB, mxc, atoms, cn, j, j)
+        C6jj = getc6(C6AB, mxc, charges, cn, j, j)
 
-        z = atoms[j]
+        z = int(charges[j])
```

Comment:
We can keep charges as floating point numbers throughout. Actually, that is what is normally done in all quantum chemistry codes.

Reply:
it has to change to int so that it cab get the correct parameters
`z = int(charges[j])`
`C8jj = 3.0 * C6jj * math.pow(R2R4[z], 2.0)`

Reply:
I see!